std/xml.d

```diff
@@ -2714,6 +2715,16 @@ EOS";
 
 unittest
 {
+    string test_xml = `<?xml version="1.0" encoding='UTF-8'?><stream:stream
+                        xmlns:stream="http://etherx.'jabber'.org/streams"
+                        xmlns="jabber:'client'" from='jid.pl' id="587a5767"
+                        xml:lang="en" version="1.0"></stream:stream>`;
+
+    DocumentParser parser = new DocumentParser(test_xml);
```

Comment:
> Should this also test that the parsed structure matches the expected structure? I.e., that the value of encoding is correctly parsed as UTF-8?

Yeah, that should be tested. Unfortunately, std.xml doesn't seem to generate events for the root element, making this a bit more cumbersome than it should be.

@coderabhishek, if your motivation is still up for it, please add `assert`s for the values of the attributes. You can see how to do that in [the `unittest` block that's below the one you added](https://github.com/coderabhishek/phobos/blob/63eae7ea322863a3a90268e0868589c60278ebd9/std/xml.d#L2726-L2743). Like there, you'll have to wrap things in a new root element. Also include a case like `attr='a"b"c'`.